### DISCUSSION TITLE
Aligning recorded streams to the hour

### DISCUSSION THREAD

**Discussion by diebels** [bluenviron/mediamtx#3571] [Questions]
### Question

Hi, I'm using mediamtx-1.6.0 and recording RTSP streams with 1 hour duration "recordSegmentDuration: 1h". Is there any way to align these to the hour other than restarting the mediamtx service at the hour? Has there been any feature added for this in newer releases? I couldn't find anything to indicate that in changelogs.

**Comment by aler9** [comment]
Hello, you can use the [Playback server](https://github.com/bluenviron/mediamtx?tab=readme-ov-file#playback-recorded-streams) to download recordings that are precisely aligned to the hour:

```
http://localhost:9996/get?path=mypath&start=2024-01-14T16%3A00%3A00%2B00%3A00&duration=3600&format=mp4
```

The playback server is the interface of choice for downloading recordings with a certain start date, duration and format, and works by cutting and merging underlying recording segments, which are the files stored on disk.

Conversely, recording segments are designed for robustness in case of power losses, not for direct usage (although they can be played). Aligning them to the hour would mean introducing an additional work load that is already present, on demand, inside the playback server.
